kernel-spark/src/main/java/io/delta/kernel/spark/read/SparkMicroBatchStream.java

```
@@ -57,19 +88,92 @@ public SparkMicroBatchStream(String tablePath, Configuration hadoopConf) {
 
   @Override
   public Offset initialOffset() {
+    // TODO(#5318): Implement initialOffset
     throw new UnsupportedOperationException("initialOffset is not supported");
   }
 
   @Override
   public Offset latestOffset() {
-    throw new UnsupportedOperationException("latestOffset is not supported");
+    // TODO(#5318): Implement latestOffset with proper start offset and limit
```

Can we just throw InvalidStateException with error msg "latestOffset() should not be called - use latestOffset(Offset, ReadLimit) instead"

---

Done.